### PR TITLE
feat(vis): Events for timeline-chart

### PR DESCRIPTION
**Pull request by marcelorafaelfeil**
- **What feature or fix was requested?**<br/>
*Implement tooltip for timelinechart.*<br/>
 
- **Is there any specific condition to reproduce the scenario?**<br/>
*It's necessary or return info property inside data from the backend, or change change the tooltip render with onRenderTimelineChartTooltip.*<br/>

- **Which functions have you tested your implementation?**<br/>
*Demo*<br/>

There is 1 PR depending on these changes:
https://github.com/philips-emr/tasy-framework/pull/14090

### CODE REVIEW

File: lib/DOMutil.js [hunk 1]
```diff
@@ -270,6 +270,34 @@ exports.drawPoint = function(x, y, groupTemplate, JSONcontainer, svgContainer, l
...
+      } else {
+        ele.addEventListener(name, (eve) => callback(eve, ele, data[index]));
+      }
+    });
+  } else {
+    element.addEventListener(name, (eve) => exports.eventCallback(eve, element, data));
+  }
+}
+
+exports.detachEvents = function(element, name, event) {
```
Comment (leohammes) [inline lib/DOMutil.js:287]: Where is this **detachEvents** method being called from?

File: lib/timeline/component/timeline-chart/TimelineChartLineGraph.js [hunk 2]
```diff
@@ -173,6 +193,50 @@ class TimelineChartLineGraph extends LineGraph {
     return false;
   }
 
+  setOptions(options) {
```
Comment (leohammes) [inline lib/timeline/component/timeline-chart/TimelineChartLineGraph.js:196]: Is this **setOptions** method needed to do some specific behavior?

**Review by leohammes — CHANGES_REQUESTED** [review]

File: lib/timeline/component/timeline-chart/TimelineChartLineGraph.js (hunk 2 above)
Comment (marcelorafaelfeil) [reply lib/timeline/component/timeline-chart/TimelineChartLineGraph.js:196]: Yes. When the componente is registered (TimelineChart.js:84), the parent class read the variable this.componentes and invoke some methods. One of them is the setOptions. The Core.js pass automatically the options through the componente and the setOptions filter only the properties specified at fields (TimelineChartLineGraph.js:198).

File: lib/DOMutil.js [hunk 3]
```diff
@@ -270,6 +270,20 @@ exports.drawPoint = function(x, y, groupTemplate, JSONcontainer, svgContainer, l
...
+  if (element && Array.isArray(element)) {
+    element.forEach((ele, index) => {
+      if (Array.isArray(ele)) {
+        ele.forEach(e => e.addEventListener(name, (eve) => callback(eve, e, data[index])));
+      } else {
+        ele.addEventListener(name, (eve) => callback(eve, ele, data[index]));
+      }
+    });
+  } else {
+    element.addEventListener(name, (eve) => exports.eventCallback(eve, element, data));
```
Comment (leohammes) [inline lib/DOMutil.js:283]: exports is undefined

**Review by leohammes — CHANGES_REQUESTED** [review]

**Pull request closed without merging**